File: 1437. Check If All 1's Are at Least Length K Places Away/Solution.py
class Solution:
    def kLengthApart(self, nums: List[int], k: int) -> bool:
        locations = []
        for i,v in enumerate(nums):
            if v == 1:
                locations.append(i)
        
        for i in range(len(locations) - 1):
            if (locations[i+1] - locations[i]) < (k + 1):
                return False
            
        return True
        

File: 1437. Check If All 1's Are at Least Length K Places Away/Solution.cpp
class Solution {
public:
    bool kLengthApart(vector<int>& nums, int k) {
        
        std::vector<int> pos;
        
        int cnt = 0;
        for(std::vector<int>::iterator it = nums.begin() ; it != nums.end(); ++it)
        {
            if(*it == 1)
            {
                pos.push_back(cnt);
            }
            cnt++;
        }
        
        if (pos.size() == 0)
            return true;
        
        for(int i=0;i < pos.size() - 1;i++)
        {
            if(pos[i+1] - pos[i] < (k + 1))
            {
                return false;
            }
        }
        
        return true;
        
    }
};
